src/operator/nn/upsampling-inl.h

```diff
@@ -84,6 +86,23 @@ struct UpSamplingParam : public dmlc::Parameter<UpSamplingParam> {
   }
 };  // struct UpSamplingParam
 
+inline int* scaleComp(const UpSamplingParam &param) {
```

Comment:
maybe return a std::vector<int>? so you don't need to new/free the dynanic array.

Reply:
Good idea--will make changes